config/webpack.config.js

```diff
@@ -197,6 +284,50 @@ const configGenerator = (buildOptions, apps) => {
         fileName: 'file-manifest.json',
       }),
     );
+  } else {
+    // Add landing pages for local development with the watch task
+    baseConfig.plugins = baseConfig.plugins.concat(
+      getAppManifests()
+        .map(m => {
+          // There's no place to create a landing page at
+          if (!m.rootUrl) return undefined;
+
+          const landingPagePath = path.join(
+            outputPath,
+            '../',
+            m.rootUrl,
+            'index.html',
+          );
+
+          // Only create a new landing page if one doesn't already exist from a
+          // previous build. This is useful for using the content build page for
+          // testing.
+          if (fs.existsSync(landingPagePath)) return undefined;
+
+          return new HtmlWebpackPlugin({
+            filename: landingPagePath,
+            template:
+              m.landingPageDevTemplate ||
+              'src/platform/landing-pages/dev-template.ejs',
+            // Pass data to the tempates
+            templateParameters: {
+              // Everything from the manifest file
+              ...m,
+              // With some defaults
+              loadingMessage:
+                m.loadingMessage ||
+                'Please wait while we load the application for you.',
+              entryName: m.entryName || 'static-pages',
+              // TODO: Get this placeholder data from another file
+              headerFooterData,
+            },
+            // Don't inject all the assets into all the landing pages
+            // The assets we want are referenced in the template itself
+            inject: false,
+          });
+        })
+        .filter(p => p),
```

Comment:
What is this filter doing exactly?

Reply:
When a manifest file either doesn't have a `rootUrl` specified _or_ there's already a landing page at the `rootUrl` (say, from a content build), the `.map()` will return `undefined`. I didn't want to add `undefined` items to the `baseConfig.plugins` array, so this filter just keeps everything that's truthy.

Reply:
Should filter before mapping here, too.